### PR TITLE
Mongoose improvements

### PR DESCRIPTION
**Pull request by allanchau**
This PR updates mongoose to require passing in a mongoose instance. This ensures the mongoose instances are the same between this library and the one in the app.

## Notes

- The parameter structure has been updated to be an object with the following keys:
  - mongoose
  - options = {}
  - uri/uris

### Testing

- [x] After updating the parameters, everything should work as it did before.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -334,6 +339,15 @@
     eslint "^5.0.1"
     eslint-plugin-eslint-comments "^3.0.0"
 
+"@idearium/log@1.0.0-beta.2":
```
Comment (smebberson) [inline yarn.lock:342]: @allanchau, this doesn't seem right? It should be `beta.6`?

**Review by smebberson — CHANGES_REQUESTED** [review]
@allanchau, I found something strange...

File: yarn.lock (hunk 1 above)
Comment (allanchau) [reply yarn.lock]: @smebberson I've updated it now, but it shouldn't really matter because it isn't used? We could probably add it to gitignore.

Comment (smebberson) [reply yarn.lock]: What do you mean? It's totally used? See https://github.com/idearium/idearium-lib/pull/90/files#diff-6bbe3edcb5cef2c785a41583b1977e9b5487596b908cefcb6b1a13696b77664aL4? Am I missing something?

Comment (allanchau) [reply yarn.lock]: @smebberson I mean the yarn.lock file in the root of the monorepo isn't deployed with the packages? It shouldn't affect how it is used? I think as long as the package.json in the packages is correct it should be ok

**Review by smebberson — CHANGES_REQUESTED** [review]
@allanchau, there are some conflicts here. Could you please tidy this up and while you're there update the @idearium/log dependency to v1.0.0?

**Review by smebberson — APPROVED** [review]

**Pull request merged**